src/EditorFeatures/Core/StackTraceExplorer/StackTraceAnalyzer.cs

```diff
@@ -48,45 +49,13 @@ private static IEnumerable<ParsedFrame> Parse(string callstack)
 
         private static IEnumerable<string> SplitLines(string callstack)
         {
-            // ActivityLog.xml line split
-            if (callstack.Contains("&#x000D;&#x000A;"))
-            {
-                return SplitLines(callstack, "&#x000D;&#x000A;");
-            }
+            // if the callstack comes from ActivityLog.xml it has been
+            // encoding to be passed over HTTP. This should only decode 
+            // specific characters like "&gt;" and "&lt;" to their "normal"
+            // equivalents ">" and "<" so we can parse correctly
+            callstack = WebUtility.HtmlDecode(callstack);
```

Comment:
Don't hate me, but, just for some fun test data...
![image](https://user-images.githubusercontent.com/754264/135027672-a7553b40-0c6c-45b0-adf9-c2726c915041.png)

Reply:
I don't hate you more than I did before this comment ;) 

Agree there could be cases where we decode incorrectly, but I think the function lookup would still work as intended. The parameter values would display wrong... If it gets enough traction maybe we'll parse differently. I'm this close |-| to just writing a manual parser for this instead of using Regex...

Reply:
1. Write regex to parse frames
2. Turn on regex source generator in .NET 6
3. Open the generated code in VS
4. Copy the code into Roslyn and manually edit it so it does only what you need
5. ???
6. Profit

Reply:
I thought they were in .NET 7??? https://github.com/dotnet/runtime/pull/59186